scmrepo/git/__init__.py

```diff
@@ -397,3 +397,15 @@ def stash_workspace(self, **kwargs):
 
     def _reset(self) -> None:
         self.backends.reset_all()
+
+    def describe(
+        self,
+        rev: str,
+        base: Optional[str] = None,
+        match: Optional[str] = None,
+        exclude: Optional[str] = None,
+    ) -> Optional[str]:
+        if base == "refs/heads":
+            if self.get_rev() == rev:
+                return base + "/" + self.active_branch()
```

Comment:
This won't always work because `HEAD` does always point to a branch (meaning `active_branch()` will raise an exception). (You can test what happens in this case by doing `git checkout` with a tag name or an explicit commit SHA) and then trying to `describe()` that git SHA

Reply:
Only happened with explicit commit SHA, for tags they have higher priority and not go into `if base == "refs/heads"` branch